ESSArch_Core/util.py

```diff
@@ -737,3 +738,17 @@ def zip_directory(dirname=None, zipname=None, compress=False):
                 filepath = os.path.join(root, f)
                 arcname = os.path.relpath(filepath, dirname)
                 new_zip.write(filepath, arcname)
+
+
+def has_write_access(directory):
+    if os.name == 'nt':
+        try:
+            tmp_file = os.path.join(directory, str(uuid.uuid4()))
```

Comment:
To minimize the risk of the file not being deleted, we should use tempfile.TemporaryFile as a context manager which automatically deletes the file on completion , e.g.

```
with tempfile.TemporaryFile(mode='a', dir=directory) as fp:
    fp.write('')
```

Reply:
Added a comment for the workaround instead. :)